Scan and monitor examples using libgpiod

fsheikh opened the pull request:
Summary: This patch adds a simple C example to search for a GPIO line
on a Linux platform with multiple GPIO controllers. It also adds another
c++ example using event loop to monitor a GPIO line.

fsheikh commented:
Hi @starnight. I really found the examples quite useful. Although one can look into ligpiod official project, they are still useful for example when one does not want to rebuilt that whole project. Just added some more examples which fitted a use-case I was working on. All in the hope that others may find them useful. Please feel free to review. Cheers!

starnight commented on libgpiod-monitor/main.cpp:
```diff
@@ -0,0 +1,211 @@
+#include <unistd.h>
+
+
+// constants
```
Please transfer whole line comment from `//` to `/* ... */` for the coding style.  Include the following line comments.

starnight commented on libgpiod-monitor/main.cpp:
```diff
@@ -0,0 +1,211 @@
+    unsigned int numLine = 0;
+
+    // Boilerplate command line arguments processing
+    while((optIn = getopt(argc, argv, "uhn:l:")) != EOF) {
```
I did not know `getopt()` before.  I learned this is useful function!  Cool!  Thanks :)

starnight reviewed (commented):
Thanks for your contribution.  However, please split the two functions/folders: libgpiod-monitor and libgpiod-scan from one commit to two commits.  Each commit for one of the function.

fsheikh committed:
libgpiod-scan: New example to for scanning devices

Summary: This examples supports a libgpiod client which scans
available devices/controllers on a platform to find which of
these supports a given GPIO line.

Signed-off-by: Faheem Sheikh <fahim.sheikh@gmail.com>

fsheikh committed:
libgpiod-monitor: C++ example monitoring gpio line

Summary: This patch presents a C++ example (without using
libgpiod c++ bindings) to monitor events on a given GPIO
line.

Signed-off-by: Faheem Sheikh <fahim.sheikh@gmail.com>

fsheikh commented:
@starnight please feel free to have another look!

starnight commented on libgpiod-monitor/main.cpp:
```diff
@@ -0,0 +1,211 @@
+    std::cout << std::string(progName) << std::endl;
+    std::cout << "     -u/h  : Usage message " << std::endl;
+    std::cout << "     -n    : Number of GPIO controller to scan, Default range [0,1)" << std::endl;
+    std::cout << "     -l    : Line number to search in GPIO controllersi, Default 0" << std::endl;
```
Is "controllersi" typo?

fsheikh replied:
Done!

starnight commented on libgpiod-scan/main.c:
```diff
@@ -0,0 +1,92 @@
+    printf("%s: \n", progName);
+    printf("     -u/h  : Usage message \n");
+    printf("     -n    : Number of GPIO controller to scan, Default range [0,1) \n");
+    printf("     -l    : Line number to search in GPIO controllersi, Default 0 \n");
```
Is "controllersi" typo?

fsheikh replied:
Done!

starnight commented on libgpiod-scan/main.c:
```diff
@@ -0,0 +1,92 @@
+    int numDevices = -1;
+    int numLine = -1;
+
+    // Boilerplate command line arguments processing
```
Please use the comment as /* ... */ here.

fsheikh replied:
Done!

starnight commented on libgpiod-scan/main.c:
```diff
@@ -0,0 +1,92 @@
+
+    // Boilerplate command line arguments processing
+    while((optIn = getopt(argc, argv, "uhn:l:")) != EOF) {
+        printf("getopt loop running\n");
```
Is here any special idea to put this `printf`?

fsheikh replied:
No good reason, possibly left-over from a debug-cycle, removed!

starnight reviewed (changes requested):
After 4 years, I just learned I did not submit the review request.

fsheikh committed:
Incorporate review comments

Summary: Addressed review comments four years after.

fsheikh commented:
> After 4 years, I just learned I did not submit the review request.

Perhaps it is never too late :smile:

starnight commented:
Squashed the fix into the previous commits and pushed to master.  Thanks
